src/OrbitQt/TrackTypeItemModelTest.cpp

```diff
@@ -0,0 +1,102 @@
+// Copyright (c) 2021 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include <gtest/gtest.h>
+
+#include <QAbstractItemModelTester>
+#include <QModelIndex>
+#include <QSignalSpy>
+#include <QTableView>
+#include <QTest>
+#include <string>
+
+#include "QtUtils/AssertNoQtLogWarnings.h"
+#include "TimeGraphLayout.h"
+#include "TrackManager.h"
+#include "TrackManagerTest.h"
+#include "TrackTypeItemModel.h"
+
+namespace orbit_qt {
+
+class TrackTypeItemModelTest : public ::testing::Test {
+ public:
+  TrackTypeItemModelTest()
+      : Test(),
+        capture_data_(orbit_gl::TrackTestData::GenerateTestCaptureData()),
+        track_manager_(nullptr, nullptr, &layout_, nullptr, capture_data_.get()) {}
+
+  int FindRowByTrackType(Track::Type track_type) {
+    int found_row = -1;
+
+    for (int i = 0; i < model_.rowCount(); ++i) {
+      Track::Type this_row_type =
+          model_.data(model_.index(i, 0), Qt::UserRole).value<Track::Type>();
+      if (track_type == this_row_type) {
+        found_row = i;
+      }
+    }
+
+    EXPECT_TRUE(found_row != -1);
+    return found_row;
+  }
+
+ protected:
+  // This installs a QtMessageHandler for this scope. Any warning, critical or fatal message
+  // produced by Qt in this scope will produce a GTest fail assertion. (Debug and info messages are
+  // printed, but do not lead to a fail). In this scope QAbstractItemModelTester is used to
+  // automatically test the ProcessItemModel. This QAbstractModelTester produces these messages and
+  // AssertNoQtLogWarnings is necessary to bridge a Qt message to a GTest failure.
+  orbit_qt_utils::AssertNoQtLogWarnings log_qt_test_;
+  TrackTypeItemModel model_;
+
+  TimeGraphLayout layout_;
+  std::unique_ptr<orbit_client_model::CaptureData> capture_data_;
+  TrackManager track_manager_;
+};
+
+TEST_F(TrackTypeItemModelTest, QtBasicTests) {
+  QAbstractItemModelTester tester(&model_, QAbstractItemModelTester::FailureReportingMode::Warning);
+}
+
+TEST_F(TrackTypeItemModelTest, ReadAndWriteData) {
+  EXPECT_EQ(0, model_.rowCount());
+
+  // When a track manager is set, the data should be non-empty
+  model_.SetTrackManager(&track_manager_);
+  EXPECT_GT(model_.rowCount(), 0);
+
+  // Search for the "ThreadTrack" entry and make sure it exists
+  // This is an arbitrary example of a track type that we expect to be in the model
```

Comment:
```suggestion
  // Search for the "ThreadTrack" entry and make sure it exists.
  // This is an arbitrary example of a track type that we expect to be in the model.
```
At least when it's two sentences in a row.